src/main/java/mods/eln/server/ServerEventListener.java

```diff
@@ -120,32 +121,31 @@ public void onWorldSave(Save e) {
             NBTTagCompound nbt = new NBTTagCompound();
             writeToEaWorldNBT(nbt, e.world.provider.dimensionId);
 
-            String worldSaveName = getEaWorldSaveName(e.world);
-            String tempSaveName = worldSaveName + ".tmp";
-            String backupSaveName = worldSaveName + ".bak";
-            File failedSave = new File(tempSaveName);
-            if (failedSave.exists()) {
-                failedSave.delete();
-            }
+            FileNames fileNames = new FileNames(e);
 
-            FileOutputStream fileStream = new FileOutputStream(tempSaveName);
-            CompressedStreamTools.writeCompressed(nbt, fileStream);
-            fileStream.flush();
-            fileStream.close();
+            // Write a new save to a temporary file.
+            final ByteArrayOutputStream bytes = new ByteArrayOutputStream(512 * 1024);
+            CompressedStreamTools.writeCompressed(nbt, bytes);
+            Files.write(fileNames.tempSave, bytes.toByteArray());
 
-            new File(worldSaveName).renameTo(new File(backupSaveName));
-            new File(tempSaveName).renameTo(new File(worldSaveName));
+            // Replace backup save with old save, and old save with new one.
+            replaceFile(fileNames.worldSave, fileNames.backupSave);
+            replaceFile(fileNames.tempSave, fileNames.worldSave);
         } catch (Exception ex) {
             ex.printStackTrace();
         }
+    }
 
-        //ElnWorldStorage storage = ElnWorldStorage.forWorld(e.world);
-        int idx = 0;
-        idx++;
+    private void replaceFile(Path from, Path to) throws IOException {
+        try {
+            Files.move(from, to, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
Just read the doc. It is not clear:

> The move is performed as an atomic file system operation and all other options are ignored. If the target file exists then it is implementation specific if the existing file is replaced or this method fails by throwing an IOException.

Again implementation specific 🙄. If we take phrase one, then you are right, REPLACE_EXISTING is not needed.